libs/wallet/src/web3-react/updater.ts

```diff
@@ -86,19 +90,39 @@ function _useSafeInfo(walletInfo: WalletInfo): GnosisSafeInfo | undefined {
       }
     }
 
-    const interval = setInterval(update, SAFE_INFO_UPDATE_INTERVAL)
+    const updateSafeAppsSdkInfo = async () => {
+      if (safeAppsSdk) {
+        const appsSdkSafeInfo = await safeAppsSdk.safe.getInfo()
+        setSafeInfo((prevSafeInfo) => {
+          // only update the isReadyOnly flag if we have safe info returned from the api kit
+          if (!prevSafeInfo) return
+          return {
+            ...prevSafeInfo,
+            isReadOnly: appsSdkSafeInfo.isReadOnly,
+          }
```

Comment:
Small optimization idea: only update if values are different:

```suggestion
          if (!prevSafeInfo || prevSafeInfo.isReadOnly === appsSdkSafeInfo.isReadOnly) return

          return {
            ...prevSafeInfo,
            isReadOnly: appsSdkSafeInfo.isReadOnly,
          }
```

Reply:
I thought about this, but then decided it's not worth it. React shouldn't trigger a re-render if the state is not changing...